### PR TITLE
Prototype identity adapter

### PR DESCRIPTION
**Pull request by joheredi**

### CODE REVIEW

File: test/azure/generated/AzureSpecials/autoRestAzureSpecialParametersTestClientContext.ts [hunk 1]
```diff
@@ -10,6 +10,9 @@
 import * as Models from "./models";
 import * as msRest from "@azure/ms-rest-js";
 import * as msRestAzure from "@azure/ms-rest-azure-js";
+import { TokenCredential } from "@azure/core-auth";
+import { AzureIdentityCredentialAdapter } from "./azureIdentityCredentialAdapter";
+import { isTokenCredential } from "@azure/core-auth";
```
Comment (ramya-rao-a) [inline test/azure/generated/AzureSpecials/autoRestAzureSpecialParametersTestClientContext.ts:15]: nit: combine this with line 13

File: test/azure/generated/AzureSpecials/autoRestAzureSpecialParametersTestClientContext.ts [hunk 2]
```diff
@@ -26,7 +29,7 @@ export class AutoRestAzureSpecialParametersTestClientContext extends msRestAzure
    * credentials. The value is always '1234-5678-9012-3456'
    * @param [options] The parameter options
    */
-  constructor(credentials: msRest.ServiceClientCredentials, subscriptionId: string, options?: Models.AutoRestAzureSpecialParametersTestClientOptions) {
+  constructor(credentials: msRest.ServiceClientCredentials | TokenCredential, subscriptionId: string, options?: Models.AutoRestAzureSpecialParametersTestClientOptions) {
```
Comment (ramya-rao-a) [inline test/azure/generated/AzureSpecials/autoRestAzureSpecialParametersTestClientContext.ts:32]: Can we have the jsdocs generated above for `credentials` to add a hint on how to obtain the credentials?

File: test/azure/generated/AzureSpecials/azureIdentityCredentialAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  ServiceClientCredentials,
+  Constants as MSRestConstants,
+  WebResource,
+} from "@azure/ms-rest-js";
+import { TokenResponse } from "@azure/ms-rest-nodeauth/dist/lib/credentials/tokenClientCredentials";
+
+const DEFAULT_AUTHORIZATION_SCHEME = "Bearer";
+
+/**
+ * This class provides a simple extension to use {@link TokenCredential} from com.azure:azure-identity library to
```
Comment (ramya-rao-a) [inline test/azure/generated/AzureSpecials/azureIdentityCredentialAdapter.ts:12]:
```suggestion
 * This class provides a simple extension to use {@link TokenCredential} from @azure/identity library to
```

File: test/azure/generated/AzureSpecials/azureIdentityCredentialAdapter.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * use with legacy Azure SDKs that accept {@link ServiceClientCredentials} family of credentials for authentication.
+ */
+export class AzureIdentityCredentialAdapter
+  implements ServiceClientCredentials {
+  private azureTokenCredential: TokenCredential;
+  private scopes: string | string[];
+  constructor(
+    azureTokenCredential: TokenCredential,
+    scopes: string | string[] = "https://management.azure.com/.default"
```
Comment (ramya-rao-a) [inline test/azure/generated/AzureSpecials/azureIdentityCredentialAdapter.ts:21]: As discussed offline, the generator should allow passing of scopes

**Review by ramya-rao-a — COMMENTED** [review]
Looks good! Left a few comments

**Comment by joheredi** [comment]
Closing in favor of #846

**Pull request closed without merging**